packages/next-app/src/app/stableswapsimulator/(components)/TokenForm.tsx

```diff
@@ -38,10 +38,13 @@ export default function TokenForm({
     resolver: zodResolver(stableSwapTokensSchema),
   });
 
-  const currentToken =
-    baselineData?.tokens?.find(
-      (token: TokensData) => token.symbol == symbolToEdit
-    ) || undefined;
+  const currentToken = variant
+    ? variantData?.tokens?.find(
+        (token: TokensData) => token.symbol == symbolToEdit
+      ) || undefined
+    : baselineData?.tokens?.find(
+        (token: TokensData) => token.symbol == symbolToEdit
+      ) || undefined;
```

Comment:
Since you're doing the same operation, you could do something like the follow to make it easier to parse:
```suggestion
  const tokens = variant ? variantData : baselineData
  const currentToken = tokens?.find(
        (token: TokensData) => token.symbol == symbolToEdit
      ) 
```